daphne/src/vdaf/mod.rs

```diff
@@ -964,14 +991,17 @@ impl VdafConfig {
     /// * `state` is the helper's current state.
     ///
     /// * `agg_cont_req` is the aggregate request sent by the Leader.
-    pub(crate) fn handle_agg_job_cont_req(
-        &self,
-        task_id: &TaskId,
-        agg_job_id: &MetaAggregationJobId,
+    pub(crate) fn handle_agg_job_cont_req<'s>(
+        &'s self,
+        task_id: &'s TaskId,
+        agg_job_id: &'s MetaAggregationJobId,
         state: DapHelperState,
-        agg_cont_req: &AggregationJobContinueReq,
-        metrics: &ContextualizedDaphneMetrics<'_>,
-    ) -> Result<DapHelperTransition<AggregationJobResp>, DapAbort> {
+        agg_cont_req: &'s AggregationJobContinueReq,
+        metrics: &'s ContextualizedDaphneMetrics<'s>,
+    ) -> Result<
+        impl Iterator<Item = Result<(Transition, Option<DapOutputShare>), DapAbort>> + 's,
```

Comment:
* Are we handling the `DapAbort` properly? If the result of `handle_agg_job_cont_req()` is `Err`, then what we actually want to abort the request.
* An alternative to iterating over `Option<DapOutputShare>` is to merge all of the `DapOutputShare`s returned by this function into a single `DapAggregateShare` and return that isntead.